src/pybind/mgr/cephadm/module.py

```diff
@@ -2419,7 +2419,8 @@ def list_daemons(self,
                      daemon_type: Optional[str] = None,
                      daemon_id: Optional[str] = None,
                      host: Optional[str] = None,
-                     refresh: bool = False) -> List[orchestrator.DaemonDescription]:
+                     refresh: bool = False,
+                     status: Optional[List[str]] = None) -> List[orchestrator.DaemonDescription]:
```

Comment:
These are supposed to matching `running`, `error`, `stopped` right? Do we want to validate whatever is passed in this list is actually a potential status? We know if someone was to pass `foo` or whatever for the status we're guaranteed to get no daemons.

Reply:
yup, you are right. Is there an interface or enum which maintains all the possible daemon status?

Reply:
Yes, there is a `DaemonDescriptionStatus` class that lives in mgr/orchestrator/_interface.py that specifies them. You should be able to build a list of the status names to check against from that class. For example, in a little test script containing
```
import enum

from typing import Optional

class DaemonDescriptionStatus(enum.IntEnum):
    unknown = -2
    error = -1
    stopped = 0
    running = 1
    starting = 2  #: Daemon is deployed, but not yet running

    @staticmethod
    def to_str(status: Optional['DaemonDescriptionStatus']) -> str:
        if status is None:
            status = DaemonDescriptionStatus.unknown
        return {
            DaemonDescriptionStatus.unknown: 'unknown',
            DaemonDescriptionStatus.error: 'error',
            DaemonDescriptionStatus.stopped: 'stopped',
            DaemonDescriptionStatus.running: 'running',
            DaemonDescriptionStatus.starting: 'starting',
        }.get(status, '<unknown>')


print('foo' in [s.name for s in DaemonDescriptionStatus])
print('running' in [s.name for s in DaemonDescriptionStatus])
```
gives me
```
bash-5.2$ python3 daemon-status-get-valid-status.py 
False
True
```